File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
# 식 입력
ipt = input()

# 숫자, 연산자 분리
nums = []
num = ""
operator = []

for i in ipt:
    if i == "+" or i == "-":
        operator.append(i)
        nums.append(num)
        num = ""
    else:
        num += i
nums.append(num)
nums = list(map(int, nums))

# (-) 부호 인덱스 찾기
try:
    idx = operator.index("-")
except:
    idx = 100

# 계산
cal = 0
for i in range(len(nums)):
    if i <= idx:
        cal += nums[i]
    else:
        cal -= nums[i]

# 출력
print(cal)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.kt
fun main() {
    val input = readln()
    var ans = 0
    var num = 0
    var minus = false
    for (c in input) {
        if (c in '0'..'9') num = num * 10 + (c - '0')
        else {
            if (minus) ans -= num else ans += num
            if (c == '-') minus = true
            num = 0
        }
    }
    if (minus) ans -= num else ans += num
    println(ans)
}